### ISSUE TITLE
If the pull request already exists, update the description, and open it in the browser if -o is given

### ISSUE THREAD

**Issue by axos88** [github/hub#1650]
Repro:
  1. `git pull-request -b development -m "OLD MSG"`  # Pull request is created
  2. `git pull-request -b development -o` # Invoke again

Expected:
 - the current PR description should be shown in the editor with the possibility to modify it and then update the description of the PR
  - the URL for the preexisting PR should be opened in the browser.

**Comment by mislav** [comment]
Interesting. I'm not sure if we'd want to implement feature 1 (it might violate the principle of least surprise) but feature 2 is definitely interesting if `-o` was passed. Keeping open as something to think about.

**Comment by axos88** [comment]
I don't think it would come as a surprise, if the current PRs description, etc. would be shown to the user. The top of the file could also contain a comment with the fact that this PR already exists. It could also be put behind a new option or just ask the user for confirmatoin - PR already exists - edit instead?

**Comment by realyze** [comment]
This would be really great. I'm using `hub` to make it easier to create and submit multiple chained PRs (e.g. to make reviews easier by splitting a feature into more manageable chunks) and it would be splendid if I could edit the PR descriptions with links to the other PRs in the chain.

**Comment by igbanam** [comment]
I think this would be an amazing addition.

```If a PR exists on GitHub, get the current PR description and use that as the text in the editor. If a PR does not exist, use the current PR template generated```

Twould be nice if we got this functionality just by using `hub pull-request`

Other things to watch out for is the assignments of labels, reviewers, and assignees. I believe these should be added on to the ones already existing in the PR

**Comment by bxt** [comment]
This is somehow related to #867, too.

I think having a command like `git pull-request --upsert` like @arcanis suggested above would solve all the issues. It creates a PR or updates the PR if it exists, does not fail, and still returns the PR's URL.

That would also help a lot in scripts that automatically create PRs.

**Comment by ARXChrono** [comment]
Any progress on this issue here?

I'm actually looking to automate my pull request but the closest thing i could find to check if the PR exist already is using `hub pr list` which kinda gets my half way. It will still crash when it tries to open a PR if one already exist.

**Comment by mislav** [comment]
@ARXChrono You can use `hub pr list` to query PRs based on different criteria or `hub pr show` to list an existing PR that belongs to the current branch. You can use the latter to check if a PR exists before opening a new one with `hub pull-request`.

There is still no command to update a pull request.

**Comment by ARXChrono** [comment]
@mislav I only see `hub pr list` and `hub pr checkout` , I don't think `hub pr show` is a thing anymore?

**Comment by mislav** [comment]
@ARXChrono You may need to upgrade https://github.com/github/hub/releases/tag/v2.12.0.

**Comment by mislav** [comment]
Cross-referencing an earlier, closely related issue #1523
